worker/uniter/relation/relationer.go

```diff
@@ -105,7 +110,15 @@ func (r *Relationer) CommitHook(hi hook.Info) error {
 		panic("implicit relations must not run hooks")
 	}
 	if hi.Kind == hooks.RelationBroken {
-		return r.die()
+		err := r.die()
+		if err != nil {
+			return errors.Trace(err)
+		}
+		rel := r.ru.Relation()
+		if r.suspended {
```

Comment:
I think you want to set the status first, in case something goes wrong in between. As it is, if the agent failed after die(), and then came back up, it wouldn't set status (I think?)

Reply:
moved setting the status and refactored a bit so that a failure to set status does not prevent cleanup but is still reported as an error